### PR TITLE
Document most of misc save block

### PR DESCRIPTION
**Pull request by ravepossum**
- Rename misc save file -> savedata_misc
- Rename misc save structs and make definition transparent
- Create new constants file for misc save data

### CODE REVIEW

File: include/constants/savedata/savedata_misc.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#ifndef POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+#define POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+
+#define RIVAL_NAME_LEN      7
+#define TABLET_NAME_LEN     10
+
+#endif // POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
```
Comment (cbt6) [inline include/constants/savedata/savedata_misc.h:7]: Add newline to EOF

File: include/constants/savedata/savedata_misc.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#ifndef POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+#define POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+
+#define RIVAL_NAME_LEN      7
+#define TABLET_NAME_LEN     10
```
Comment (lhearachel) [inline include/constants/savedata/savedata_misc.h:5]: There is a header `constants/string.h` that is meant to be used to store string-length constants like these.

I think `RIVAL_NAME_LEN` is thus superfluous, as it's the same length as any other trainer.

File: include/savedata_misc.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#include "struct_defs/struct_02027860.h"
+#include "savedata.h"
+#include "constants/savedata/save_table.h"
+#include "constants/savedata/savedata_misc.h"
+
+typedef struct ExtraSaveKey {
+    u32 keys[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u32 oldKeys[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u8 keyFlags[EXTRA_SAVE_TABLE_ENTRY_MAX - 1];
+    u8 filler[3];
```
Comment (lhearachel) [inline include/savedata_misc.h:18]: Convention in the project is to name member-vars like these as `padding`, with a suffix if there are multiple instances of padding in the struct.

File: include/savedata_misc.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    UnkStruct_02027854 unk_00[128];
+    UnkStruct_02027860 unk_680;
+    u16 rivalName[RIVAL_NAME_LEN + 1];
+    u16 tabletName[TABLET_NAME_LEN + 1]; // used in shaymin event
+    u16 favoriteMon;
+    u8 favoriteMonForm:7;
+    u8 favoriteMonIsEgg:1;
+    u8 extraSaveInitalized:1;
+    u8 vsRecorderColor:4;
+    u8 filler:3;
```
Comment (lhearachel) [inline include/savedata_misc.h:31]: Spaces on both sides of the `:` for bitflags:

```suggestion
    u8 favoriteMonForm : 7;
    u8 favoriteMonIsEgg : 1;
    u8 extraSaveInitalized : 1;
    u8 vsRecorderColor : 4;
    u8 filler : 3;
```

File: src/savedata_misc.c [hunk 5]
```diff
@@ -0,0 +1,143 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "strbuf.h"
+#include "savedata.h"
```
Comment (lhearachel) [inline src/savedata_misc.c:5]: Duplicate includes

**Review by lhearachel — CHANGES_REQUESTED** [review]

**Review by lhearachel — APPROVED** [review]

**Pull request merged**